lib/runtime-core/src/jit_debug.rs

```diff
@@ -0,0 +1,195 @@
+//! Code for interacting with the
+//! [GDB JIT inteface](https://sourceware.org/gdb/current/onlinedocs/gdb.html#JIT-Interface).
+
+use std::ptr;
+use std::sync::Arc;
+
+/// Entrypoint that the debugger will use to trigger a read from the
+/// [`__jit_debug_descriptor`] global variable.
+///
+/// The debugger will wait for this function to be called and then take
+/// control to read the data we prepared.
+// Implementation of this function is derived from wasmtime and is licensed under
+// the Apache 2.0 license.  See ATTRIBUTIONS.md for full license and more
+// information.
+#[no_mangle]
+#[inline(never)]
+extern "C" fn __jit_debug_register_code() {
+    // This code exists to prevent optimization of this function so that the
+    // GDB JIT interface behaves as expected
+    let x = 3;
+    unsafe {
+        std::ptr::read_volatile(&x);
+    }
+}
+
+/// The operation that the debugger should perform with the entry that we gave it.
+#[allow(non_camel_case_types)]
+#[derive(Debug)]
+#[repr(u32)]
+pub(crate) enum JITAction {
```

Comment:
```suggestion
pub(crate) enum JitAction {
```

and thus, you can remove the `#[allow(non_camel_case_types)]` directive.

Or… it should be `#[no_mangle]`d?